[Adjusted to apply to Debian's 2.6.26]

commit 1252f238db48ec419f40c1bdf30fda649860eed9
Author: Oleg Nesterov <oleg@redhat.com>
Date:   Tue Feb 16 15:02:13 2010 +0100

    x86: set_personality_ia32() misses force_personality32
    
    05d43ed8a "x86: get rid of the insane TIF_ABI_PENDING bit" forgot about
    force_personality32.  Fix.
    
    Signed-off-by: Oleg Nesterov <oleg@redhat.com>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

diff -urpN linux-source-2.6.26.orig/arch/x86/kernel/process_64.c linux-source-2.6.26/arch/x86/kernel/process_64.c
--- linux-source-2.6.26.orig/arch/x86/kernel/process_64.c	2010-02-10 01:11:15.000000000 -0700
+++ linux-source-2.6.26/arch/x86/kernel/process_64.c	2010-02-17 21:29:14.000000000 -0700
@@ -728,6 +728,7 @@ void set_personality_ia32(void)
 
 	/* Make sure to be in 32bit mode */
 	set_thread_flag(TIF_IA32);
+	current->personality |= force_personality32;
 
 	/* Prepare the first "return" to user space */
 	current_thread_info()->status |= TS_COMPAT;
